### PR TITLE
Testing script for Whiteblock

### PR DESCRIPTION
**Pull request by mhchia**
This PR includes the testing script `test/test_series.py`, which is capable of
- spinning up `n` docker nodes with `sharding-p2p-poc` running, connecting nodes, and broadcasting data
- measuring how much time it takes to receive a broadcasted data
- testing if the bootnodes functioning well

Also, added logs and fixes https://github.com/ethresearch/sharding-p2p-poc/issues/105 in `ShardManager`

#### TODO
- [x] Organize the CLI outputs
- [x] Finish the first testing logic(one shard, how long it takes to broadcast to X% nodes)

#### How to run it
```python
cd test
pip install python-dateutil
python test_series.py
```

#### Note
- The script becomes slower when the number of nodes becomes larger, i.e. 100. Should find out the reason and solve it.
- The script uses threads when spinning up nodes and performing some CLIs, to reduce the time consumed. Probably need to change it when it consumes too many resources when the number of nodes becomes larger.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by mhchia** [comment]
Note: possible TODOs
- [ ] Do not `connectShardNodes` automatically, to avoid changing the topology when subscribing to a new shard
- [ ] Add logs in `makeGeneralValidator` when nodes receive new data

**Comment by mhchia** [comment]
cc @zscole @araskachoi

File: shardmanager.go [hunk 1]
```diff
@@ -147,11 +147,11 @@ func (n *ShardManager) ListenShard(ctx context.Context, shardID ShardIDType) err
 		return err
 	}
 
-	// TODO: should set a critiria: if we have enough peers in the shard, don't connect shard nodes
-	if err := n.connectShardNodes(spanctx, shardID); err != nil {
-		logger.SetErr(spanctx, fmt.Errorf("Failed to connect to nodes in shard %v", shardID))
-		logger.Errorf("Failed to connect to nodes in shard %v", shardID)
-	}
+	// // TODO: should set a critiria: if we have enough peers in the shard, don't connect shard nodes
```
Comment (ChihChengLiang) [inline shardmanager.go]: Avoid committing commends

File: test/utils.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            encoding='utf-8',
+            **kwargs,
+        )
+
+    def cli_safe(self, cmd):
+        res = self.cli(cmd)
+        if res.returncode != 0:
+            print(res)
+            raise CLIFailure(
+                "exit_code={}, stdout={!r}, stderr={!r}".format(
```
Comment (ChihChengLiang) [inline test/utils.py]: Question: What can `!r` do?

Comment (mhchia) [reply test/utils.py]: I think it is similar to `{}`.format(repr(param))

File: shardmanager.go (hunk 1 above)
Comment (mhchia) [reply shardmanager.go]: Removed it. Thank you for pointing out!

**Comment by araskachoi** [comment]
Was wondering what the time to create 100 nodes was?

Using our script (not using docker compose), I was able to create 100 nodes + 1 boot node in about 4 minutes

start: 2018-11-26 20:52:25.593224
finish: 2018-11-26 20:56:15.300946

**Pull request merged**